src/_pytest/warning_types.py

```diff
@@ -55,6 +55,13 @@ class PytestRemovedIn8Warning(PytestDeprecationWarning):
     __module__ = "pytest"
 
 
+@final
+class PytestReturnNotNoneWarning(PytestDeprecationWarning):
```

Comment:
This should be imported in `src/pytest/__init__.py` and added to `__all__` there, since we want e.g. people to be able to ignore it via their config without having to import private API.

It should also be added to `doc/en/reference/reference.rst` with the other warnings, and probably get a new entry in `doc/en/deprecations.rst`.

Reply:
Thank you for the review, just want to be clear about the deprecations. Since we are already in 7, shall it be deprecated in 8? Or should I say 7?

Reply:
This should be a warning, not a deprecation warning

Reply:
Should it, if we plan to turn it into an error in the future?

Reply:
Deprecation warning is correct if we want to turn it into an error in the future, which seems the plan for `unittest` in the stdlib too: https://github.com/pytest-dev/pytest/issues/7337#issuecomment-1121689156.

https://github.com/python/cpython/pull/27748/files